contractcourt/channel_arbitrator.go

```diff
@@ -1348,20 +1356,65 @@ func (c *ChannelArbitrator) channelAttendant(bestHeight int32,
 				}),
 			)
 
-			// We've cooperatively closed the channel, so we're no longer
-			// needed.
+		// We've cooperatively closed the channel, so we're no longer
+		// needed.
 		case <-c.cfg.ChainEvents.CooperativeClosure:
 			log.Infof("ChannelArbitrator(%v) closing due to co-op "+
 				"closure", c.cfg.ChanPoint)
 			return
 
+		// We have broadcasted our commitment, and it is now confirmed
+		// on-chain.
+		case closeInfo := <-c.cfg.ChainEvents.LocalUnilateralClosure:
+			log.Infof("ChannelArbitrator(%v): local on-chain "+
+				"channel close", c.cfg.ChanPoint)
+
+			if c.state != StateCommitmentBroadcasted {
+				log.Errorf("ChannelArbitrator(%v): unexpected "+
+					"local on-chain channel close",
+					c.cfg.ChanPoint)
+			}
+			closeTx := closeInfo.CloseTx
+
+			contractRes := &ContractResolutions{
+				CommitHash:       closeTx.TxHash(),
+				CommitResolution: closeInfo.CommitResolution,
+				HtlcResolutions:  *closeInfo.HtlcResolutions,
+			}
+
+			// When processing a unilateral close event, we'll
+			// transition directly to the ContractClosed state.
+			// When the state machine reaches that state, we'll log
+			// out the set of resolutions.
+			stateCb := func(nextState ArbitratorState) error {
+				if nextState == StateContractClosed {
```

Comment:
If you invert this condition, you can save a level of indentation and prevent a lot of line-wrapping.
```go
if nextState != StateContractClosed {
    return nil
}
```

Admittedly you'd have to have two "return nil"s then, but on net it still seems nicer.

Reply:
Agreed! 👍